internal/sync/sync.go

```diff
@@ -215,7 +211,7 @@ var ErrRestartSync = errors.New("the sync exited because it was overloaded, rest
 
 // workEventGroup converges a group of events to prevent duplicate work.
 func (s Sync) workEventGroup(evs []timedEvent) {
-	cache := make(eventCache)
+	cache := make(eventCache, len(evs))
```

Comment:
what Add do? this seems suspicious?

Reply:
We are creating a map where we know exactly how big it is going to be in advance, so we can pre-allocate the memory for it.

Reply:
Interesting, I guess not seeing what type `eventCache` is makes this harder. I'm all down for preallocation, but we're far enough from the context that it was hard to read/guess. 

Frankly, it'd make more sense if there was a constructor that did all of this, rather than just having it at the beginning of a worker pool.